components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/HttpRequestOperation.java

```diff
@@ -141,6 +142,8 @@ private static boolean requestIsOngoing(RequestBodyChunkSubscriber bodyChunkSubs
                 if (requestLoggingEnabled) {
                     httpRequestMessageLogger.logRequest(request, nettyConnection.getOrigin());
                 }
+            } else {
+                throw new OriginUnreachableException(nettyConnection.getOrigin(), null);
```

Comment:
We use `OriginUnreachableException` specifically when a TCP connection establishment fails, and we use `TransportLostException` when an existing connection has unexpectedly drops or disconnects.

At `execute` method, we assume that `nettyConnection` is already connected. Therefore, we should emit a `TransportLostException` instead.

Reply:
Instead of throwing an exception, I would emit it with `sink.error(new TransportLostException(...)`. This makes it immediately obvious that the exception will propagate through a `Flux` pipeline instead of unwinding the stack.

OTOH this doesn't affect program correctness because `Flux` subscriber will do this for us anyway.